packages/react-devtools-shared/src/backend/renderer.js

```diff
@@ -348,10 +348,11 @@ export function getInternalReactConstants(
       case FunctionComponent:
       case IndeterminateComponent:
         return getDisplayName(resolvedType);
+      case MemoComponent:
+      case SimpleMemoComponent:
       case ForwardRef:
         return (
-          resolvedType.displayName ||
-          getDisplayName(resolvedType.render, 'Anonymous')
+          resolvedType.displayName || getDisplayName(resolvedType, 'Anonymous')
```

Comment:
In cases of `memo` and `forwardRef`, we're delegating extracting the function/component type from those HOCs to the `getDisplayName` itself. I couldn't apply recursion here because `getDisplayNameForFiber` accepts a fiber node.